### ISSUE TITLE
[Bug]: avg decimal column error when the result set is empty 

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#7282]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):ff2f95d9c31e2d29009b36b60cf9495f7c8411b6
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

ddl:
CREATE TABLE `customer` (
`c_w_id` INT NOT NULL,
`c_d_id` INT NOT NULL,
`c_id` INT NOT NULL,
`c_discount` DECIMAL(4,4) DEFAULT NULL,
`c_credit` CHAR(2) NOT NULL,
`c_last` VARCHAR(16) NOT NULL,
`c_first` VARCHAR(16) NOT NULL,
`c_credit_lim` DECIMAL(12,2) NOT NULL,
`c_balance` DECIMAL(12,2) NOT NULL,
`c_ytd_payment` DECIMAL(12,2) NOT NULL,
`c_payment_cnt` INT NOT NULL,
`c_delivery_cnt` INT NOT NULL,
`c_street_1` VARCHAR(20) NOT NULL,
`c_street_2` VARCHAR(20) NOT NULL,
`c_city` VARCHAR(20) NOT NULL,
`c_state` CHAR(2) NOT NULL,
`c_zip` CHAR(9) NOT NULL,
`c_phone` CHAR(16) NOT NULL,
`c_since` TIMESTAMP DEFAULT current_timestamp(),
`c_middle` CHAR(2) NOT NULL,
`c_data` VARCHAR(500) NOT NULL,
PRIMARY KEY (`c_w_id`,`c_d_id`,`c_id`)

mysql> SELECT c_balance                      FROM customer                      WHERE c_balance > 0.1;
Empty set (0.02 sec)

mysql> SELECT avg(c_balance)                      FROM customer                      WHERE c_balance > 0.1;
ERROR 1690 (HY000): data out of range: data type decimal128, decimal div overflow

**Comment by m-schen** [comment]
I will fix that.

**Comment by aressu1985** [comment]
fixed:60d1ea323be2a43f9c8c626e41f75ef6a6b7fa9e

mysql> CREATE TABLE customer (
    -> c_w_id INT NOT NULL,
    -> c_d_id INT NOT NULL,
    -> c_id INT NOT NULL,
    -> c_discount DECIMAL(4,4) DEFAULT NULL,
    -> c_credit CHAR(2) NOT NULL,
    -> c_last VARCHAR(16) NOT NULL,
    -> c_first VARCHAR(16) NOT NULL,
    -> c_credit_lim DECIMAL(12,2) NOT NULL,
    -> c_balance DECIMAL(12,2) NOT NULL,
    -> c_ytd_payment DECIMAL(12,2) NOT NULL,
    -> c_payment_cnt INT NOT NULL,
    -> c_delivery_cnt INT NOT NULL,
    -> c_street_1 VARCHAR(20) NOT NULL,
    -> c_street_2 VARCHAR(20) NOT NULL,
    -> c_city VARCHAR(20) NOT NULL,
    -> c_state CHAR(2) NOT NULL,
    -> c_zip CHAR(9) NOT NULL,
    -> c_phone CHAR(16) NOT NULL,
    -> c_since TIMESTAMP DEFAULT current_timestamp(),
    -> c_middle CHAR(2) NOT NULL,
    -> c_data VARCHAR(500) NOT NULL,
    -> PRIMARY KEY (c_w_id,c_d_id,c_id));
Query OK, 0 rows affected (0.01 sec)

mysql> SELECT c_balance FROM customer WHERE c_balance > 0.1;
Empty set (0.00 sec)

mysql> SELECT avg(c_balance) FROM customer WHERE c_balance > 0.1;
+----------------+
| avg(c_balance) |
+----------------+
|           NULL |
+----------------+
1 row in set (0.01 sec)
